codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -376,14 +403,10 @@ public Http2Stream open(boolean halfClosed) throws Http2Exception {
                 throw streamError(id, PROTOCOL_ERROR, "Attempting to open a stream in an invalid state: " + state);
             }
 
-            if (activeStreams.add(this)) {
-                // Update the number of active streams initiated by the endpoint.
-                createdBy().numActiveStreams++;
-
-                // Notify the listeners.
-                for (int i = 0; i < listeners.size(); i++) {
-                    listeners.get(i).onStreamActive(this);
-                }
+            if (canModifyActiveStreams()) {
```

Comment:
Humm. We could do this, but then every queued event will re-evalute the conditional via `process()`.  We check the conditional before acting on the queued events so we only have to check 1 time.